src/components/Button/ButtonBase.tsx

```diff
@@ -53,78 +54,148 @@ const getButtonStyles = (theme: Theme, type: ButtonTypes, disabled: boolean) =>
 
 const getButtonTextColor = (theme: Theme, type: ButtonTypes, disabled: boolean) => {
   const color = {
-    active: isContained(type) ? getColorOnPrimary(theme) : getColorHighEmphasis(theme),
-    disabled: isContained(type) ? getColorHighEmphasis(theme) : getColorMediumEmphasis(theme),
+    active: isContained(type) ? 'onPrimary' : 'highEmphasis',
+    disabled: isContained(type) ? 'highEmphasis' : 'mediumEmphasis',
   };
 
-  return disabled ? color.disabled : color.active;
+  return disabled ? color.disabled as keyof IColors : color.active as keyof IColors;
```

Comment:
aqui tu deveria tipar teu objeto de color, e nao forçar o tipo.
Além disso, remover IColors!!